Source/It.cs

```diff
@@ -141,5 +141,35 @@ public static string IsRegex(string regex, RegexOptions options)
 			// But evaluated every time :)
 			return Match<string>.Create(value => re.IsMatch(value), () => It.IsRegex(regex, options));
 		}
+
+		/// <summary>
+		/// Allows for setting up open generic methods on Mock Fixtures
+		/// </summary>
+		public interface AnyType
+		{
+			/// <summary>
+			/// Will contain the specific object being compared inside an open generic method.
+			/// Used for checking inside It.Is&lt;It.AnyType&gt;(x => ((int)x.Object) == 2)
+			/// </summary>
+			object Object { get; set; }
```

Comment:
Have added 2 tests to demonstrate usage, I added it for completeness in case anyone wants to add a conditional filter to the setup an open generic argument.

```
[Fact]
SetupReturns_ConditionalAnyType_MetCondition_CallsMockedReturn()
[Fact]
SetupReturns_ConditionalAnyType_UnmetCondition_DoesNotCallMockedReturn()

mock.Setup(x => x.Get<It.AnyType, It.AnyType, It.AnyType>(
	// Conditional input argument similar to any other It.Is<Type>(x => x == y)
	It.Is<It.AnyType>(type => type.Object as string == "TestInput")) 
)
	.Returns(context =>
	{
		callCount++;
		return "TestOutput";
	});
```